### ISSUE TITLE
Unable to access types other than "Storage" (ex. "Bucket") from declaration files

### ISSUE THREAD

**Issue by welkie** [googleapis/nodejs-storage#379]
#### Environment details

  - OS: Ubuntu 18.04
  - Node.js version: 8.11.4
  - npm version: 6.4.1
  - `@google-cloud/storage` version: 2.0.1

#### Steps to reproduce

  1. Create a new Node project with TypeScript
  2. Install the latest version of `google-cloud/storage`.
  3. Try to use non-`Storage` types like `Bucket` by importing them. Ex. `import { Bucket } from 'google-cloud/storage';`.

Right now, despite multiple declaration files being present in the package's `build` directory, only `index.d.ts` is set by the `package.json` as a declaration file to be made available. Because of this, most of the types are unavailable. The only type I was able to access was the `Storage` class (to instantiate a client, etc).

Here is an example repo demonstrating the issue. The command `npm build` will fail. The comments marked "compiler error" show idiomatic imports from the library, should the types be made available:

https://github.com/welkie/issue-example-ts-google-cloud-storage

Note that the TypeScript documentation on publishing states that the `types` property in `package.json` should be set to a bundle of declarations: https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

An alternative to adding a build step to concatenate the declaration files into a bundled file would be to export the imported types from the main file. For example:

```typescript
// index.ts

import { Bucket } from './bucket';

// index stuff
class Storage {};

export { Bucket, Storage };
```

That way, all types can be imported from the main file like so:

```typescript
// consumer.ts

import { Bucket, Storage } from '@google-cloud/storage';
```

**Comment by welkie** [comment]
Just to follow up, after tinkering more with it, I was able to import the non-`index.d.ts` types like so:

```typescript
import { Storage } from '@google-cloud/storage';
import { Bucket } from '@google-cloud/storage/build/src/bucket';

const client = new Storage();

const file: Bucket = {} as any;
```

However, I don't think this is the conventional way to import types. Usually, you're meant to have the types available from the package itself without specifying subdirectories.

**Comment by welkie** [comment]
I'd give them credit for doing the TS conversion to begin with. It's a big undertaking. I'm sure it will be resolved in time.

To any maintainers reading this, I recommend looking at [TypeORM's repository](https://github.com/typeorm/typeorm/blob/master/src/index.ts) as an example of a library implemented in TypeScript. You can see how they import everything into the `index.ts` file and then export it back out at the bottom, so that all types are available to consuming libraries.

**Comment by JustinBeckwith** [comment]
Thanks for the patience folks :) Fixed in #484
